framework/io/obj_io.cc

```diff
@@ -0,0 +1,392 @@
+      for (size_t cvid = 0; cvid < cell_vertices.size(); ++cvid)
+      {
+        const auto& cell_vertex = cell_vertices[cvid];
+
+        if ((bndry_vertex - cell_vertex).NormSquare() < 1.0e-12)
+        {
+          vertex_map[bvid] = cvid;
+          match_found = true;
+          break;
+        }
+      } // for cvid
+      if (not match_found)
+        throw std::logic_error(fname +
+                               ": Failed to map a boundary vertex to"
+                               "any cell vertex. Check that the edges are conformal with the "
+                               "object containing the faces.");
+    } // for bvid
+
+    // Change cell and face vertex ids to cell vertex ids
+    // using vertex map
+    for (auto& cell_ptr : block_data.at(main_block_id).cells)
+    {
+      for (uint64_t& vid : cell_ptr->vertex_ids)
+        vid = vertex_map[vid];
+
+      for (auto& face : cell_ptr->faces)
+        for (uint64_t& vid : face.vertex_ids)
+          vid = vertex_map[vid];
+    }
+
+    // Change edge vertex ids to cell vertex ids using
+    // the vertex map
+    for (size_t block_id : bndry_block_ids)
+      for (auto& edge : block_data[block_id].edges)
+      {
+        edge.first = static_cast<int>(vertex_map[edge.first]);
+        edge.second = static_cast<int>(vertex_map[edge.second]);
+      }
+  }
+  mesh->Vertices() = cell_vertices;
+  mesh->RawCells() = block_data[main_block_id].cells;
+
+  // Always do this
+  mesh->SetDimension(2);
+  mesh->Attributes() = UNSTRUCTURED;
+
+  mesh->ComputeCentroids();
+  mesh->CheckQuality();
+  mesh->BuildMeshConnectivity();
+
+  // Set boundary ids
+  if (bndry_block_ids.empty())
+  {
+    mesh->AddBoundary(0, "Default Boundary");
+  }
+  else
+  {
+    std::vector<UnpartitionedMesh::LightWeightFace*> bndry_faces;
+    for (auto& cell_ptr : mesh->RawCells())
+      for (auto& face : cell_ptr->faces)
+        if (not face.has_neighbor)
+          bndry_faces.push_back(&face);
+
+    size_t bndry_id = 0;
+    for (size_t bid : bndry_block_ids)
+    {
+      const auto& bndry_edges = block_data[bid].edges;
+
+      size_t num_faces_boundarified = 0;
+      for (const auto& edge : bndry_edges)
+      {
+        std::set<size_t> edge_vert_id_set({edge.first, edge.second});
+
+        for (auto& face_ptr : bndry_faces)
+        {
+          const auto& vert_ids = face_ptr->vertex_ids;
+          std::set<size_t> face_vert_id_set(vert_ids.begin(), vert_ids.end());
+
+          if (face_vert_id_set == edge_vert_id_set)
+          {
+            face_ptr->neighbor = bndry_id;
+            ++num_faces_boundarified;
+            break;
+          }
+        } // for face
+      }   // for edge
+
+      log.Log() << "UnpartitionedMesh: assigned " << num_faces_boundarified
+                << " faces to boundary id " << bndry_id << " with name " << block_data[bid].name;
+
+      mesh->AddBoundary(bndry_id, block_data[bid].name);
+
+      ++bndry_id;
+    } // for boundary block
```

Comment:
do we have a preference between `++a` and `a++`. should we specify a coding convention too?

Reply:
I don't think this should be in coding conventions. There are cases where you should be using pre-fix version and there cases where post-fix is just fine...

Reply:
I think the safe thing to say in our coding standard is: "prefer prefix operator for non-pod types". 99.9% of the time it won't matter which you use because the compiler is going to see what's happening and optimize appropriately. But, there are situations where it can make a difference, and, for inexperienced programmers like we're likely to have on OpenSn, it's probably worth mentioning.

Reply:
To me, this is C++ 101 and thus it does not belong into coding standards.